### DISCUSSION TITLE
replaceWithText in 5.3.201

### DISCUSSION THREAD

**Discussion by Godlike-Productions** [RomanKubiak/ctrlr#290] [Q&A] [answered]
This worked in 5.6.26 but not in 5.3.201.  

I'm just puting them up as I come across the errors. If anyone knows an answer, that's good, otherwise I'll try to track down an alternative over the next few days.

```
if fileToWrite:replaceWithText (exportJson, false, false, "\n") == false then
```

Any suggestion on replaceWithText

This is the error I get. I'm wondering if I need to move the "\n" before to be the second parameter.

At line [-1]: [C]
What: C
Namewhat: method
Name: replaceWithText
Error message: No matching overload found, candidates:
bool replaceWithText(File const&,String const&,bool,bool)

**Comment by Godlike-Productions** [comment]
Yes, that seems to work.  The parameters for this have changed order between 5.3.201 and 5.6.26

**Reply by Godlike-Productions** [reply]
Nope, spoke too soon. Still throwing errors. I need to investigate this more. It may be an initial condition issue.

**Comment by dobo365** [comment, marked as answer]
Here is what I'm using in 5.3.201. You are not using the right syntax I guess.
You can also see it in the error message that is expecting a string followed by 2 booleans.
Is _exportJson_ a string?

```
	-- Save the displayed info to a file
	f = utils.saveFileWindow ("Save Pro-1 program information to file", File(""), "*.txt", true)
	if f:isValid() then
		if f:existsAsFile() == false then
			f:create()
		end

		-- Write introduction text based on the Info type
		sTemp = "\n*** PROGRAM INFORMATION\n*** DATE: "..os.date("%Y-%m-%d").."\n\n"
--		sTemp = sTemp.."----------------------------+-----------------------------+-----------------------------+\n"
		f:replaceWithText(sTemp, false, false)

		-- Append text content of the Info listbox
		f:appendText(lbInfo:getProperty ("uiListBoxContent"), false, false)
```

replaceWithText replaces the initial content of the file
appendText appends text after the actual content of the file (in this example, the content of a listbox)

**Reply by Godlike-Productions** [reply]
Thanks, I recall trying to get this to work originally and it needed the extra parameter in 6.0.26, so yeah, just removing it is the key, I think.

Yeah, I'm trying to replace the text file. It saves the current state of the plugin.

exportJson is a local variable containing the encoded data.

The solution to the problem and some amazing advice that I failed to listen to is here - https://github.com/RomanKubiak/ctrlr/discussions/196

V6 VST work if you don't send MIDI from LUA (ie simple MIDI controls). sendMidiMessageNow() is where things seem to break.

I'll still develop with it though, because the ability to export LUA files is invaluable when working with a repo, though v 5.3.201 can import LUA files, so maybe that is a better way.
